src/components/ImportExport.js

```diff
@@ -0,0 +1,91 @@
+import React from 'react';
+import { Button } from 'material-ui';
+import tj from '@mapbox/togeojson';
+import fs from 'fs';
+import DOMParser from 'xmldom';
+import Hydrants from './../KML/Hydrants.kml';
+import KML from 'ol/format/kml';
+import Point from 'ol/geom/point';
+import Projection from 'ol/proj';
+import _ from 'lodash';
+
+
+class ImportExport extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      selectedFiles: null,
+    }
+    this.changeFile = this.changeFile.bind(this);
+    this.importFile = this.importFile.bind(this);
+  }
+
+
+  changeFile(e) {
+    this.setState({
+      selectedFiles: e.target.files
+    })
+  }
+
+
+  importFile = () =>  {
+    const { selectedFiles } = this.state;
+    const { importKMLClicked, mode } = this.props;
+    const reader = new FileReader();
+
+    reader.onload = function(event){
+      const kml = new KML().readFeatures(event.target.result)
+      const kmlMap = mode === 'trails'? processTrails(kml) : processHydrants(kml)
+      importKMLClicked(_.keyBy(kmlMap, 'id'))
+    }
+
+
+    function processTrails(klm){
```

Comment:
kml, not klm. I almost wouldn't mind if we were inconsistent and wrong everywhere, but should at least be the same, haha

Reply:
![alt text](https://apps.static-afkl.com/travel/us_en/images/klmlogo_new_tcm518-611911.jpg "klm not kml")

Reply:
But really though, I changed it to KML